OpenRA.Mods.Common/Widgets/SpriteWidget.cs

```diff
@@ -67,7 +70,14 @@ public override void Draw()
 				cachedPalette = palette;
 			}
 
-			Game.Renderer.SpriteRenderer.DrawSprite(sprite, RenderOrigin + offset, pr);
+			if (scale != cachedScale)
+			{
+				offset *= scale;
```

Comment:
If you change the scale twice (or more) without changing the sprite then the different scales will compound and break the position.